platform/core/PolyPodCoreAndroid/core/src/main/java/coop/polypoly/core/MessagePackDecoding.kt

```diff
@@ -0,0 +1,64 @@
+package coop.polypoly.core
+
+import org.msgpack.value.Value
+import org.msgpack.value.ValueFactory
+
+fun Value.getStringValue(): String? {
+    if (isNilValue) {
+        return null
+    }
+
+    return asStringValue().asString()
+}
+
+fun Value.getMapValue(): Map<Value, Value>? {
+    if (isNilValue) {
+        return null
+    }
+
+    return asMapValue().map()
+}
+
+fun Map<Value, Value>.get(key: String): Value? {
+    this[ValueFactory.newString(key)]
+}
+
+fun mapError(msgObject: Map<Value, Value>): CoreFailure {
+    val code = msgObject.get("code")?.asIntegerValue()?.asInt()
+    val message = msgObject.get("message")?.asStringValue()?.asString()
+    if (code != null && message != null) {
+        CoreExceptionCode.getByValue(code)?.also {
+            return CoreFailure(it, message)
+        }
+    }
+    throw InvalidCoreErrorFormat(msgObject.toString())
+}
+
+fun mapFeatureManifest(msgObject: Value): FeatureManifest {
```

Comment:
Next step is to have a way to automatically serialize/deserialize to/from a Kotlin model, instead of doing it by hand.